src/SumoCoders/FrameworkCoreBundle/BreadCrumb/BreadCrumbBuilder.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\BreadCrumb;
+
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\MenuItem;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+use SumoCoders\FrameworkCoreBundle\Event\ConfigureMenuEvent;
+
+final class BreadCrumbBuilder
+{
+    /**
+     * @var bool
+     */
+    private $dontExtractFromTheRoute = false;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var FactoryInterface
+     */
+    private $factory;
+
+    /**
+     * @var array
+     */
+    private $items = array();
+
+    /**
+     * @param FactoryInterface         $factory
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function __construct(FactoryInterface $factory, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->factory = $factory;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * Add items into the breadcrumb based on a given child.
+     *
+     * @param MenuItem $item
+     * @param string   $locale
+     */
+    protected function addItemsBasedOnTheChild(MenuItem $item, $locale)
+    {
+        if (null !== $item) {
```

Comment:
omdat je dan geen assignments in conditions kan krijgen. Is best-practice volgens SensioInsights